docs/spec_draft.md

```diff
@@ -941,6 +942,62 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dynamic_slice
+
+### Semantics
+
+Extracts a slice of shape `slice_sizes` from the `operand` tensor starting at
+`start_indices` and produces a `result` tensor.
+
+More formally, `result[i0, ..., iR-1] = operand[j0, ..., jR-1]` such that:
+  * `jd = effective_start_indices[d] + id`.
+  * `effective_start_indices[d] = clamp(start_indices[d], 0, dim(operand, d) - `
+    `slice_sizes[d])`.
```

Comment:
Proposed rewording:

```
More formally, `result[i0, ..., iR-1] = operand[j0, ..., jR-1]` where:
  * `jd = adjusted_start_indices[d] + id`.
  * `adjusted_start_indices = clamp(0, start_indices, shape(operand) - slice_sizes)`.
```

Feedback:
  1. Changed "effective_start_indices" to "adjusted_start_indices" to better align with the prose above.
  2. Dropped `[d]` from the second formula, because things look simpler that way.
  3. Reordered inputs to `clamp` because the correct order is: min, operand, max.

Reply:
Actually, the first bullet in the proposed rewording should read `jd = adjusted_start_indices[d][] + id`, given that `start_indices` are zero-dimensional tensors.